index.js

```diff
@@ -1,99 +1,126 @@
-const usb = require('usb')
-const yargs = require('yargs')
-const fs = require('fs')
+const usb = require("usb")
+const yargs = require("yargs")
+const fs = require("fs")
+const chalk = require("chalk")
 
+// Sleepy time function
+const sleep = async (ms) => new Promise((resolve) => setTimeout(resolve, ms))
 
 const argv = yargs
-    .option('f', {
-        alias: 'file',
-        describe: 'output video feed to file',
-        type: 'string'
-    })
-    .option('o', {
-        alias: 'stdout',
-        describe: 'send video feed to stdout for playback. eg: node index.js -o | ffplay -',
-        type: 'boolean'
-    })
-    .option('s', {
-        alias: 'readsize',
-        describe: 'size in bytes to queue for usb bulk interface reads',
-        default: 512,
-        type: 'integer'
-    })
-    .option('q', {
-        alias: 'queuesize',
-        describe: 'number of polling usb bulk read requests to keep in flight',
-        default: 3,
-        type: 'integer'
-    })
-    .option('v', {
-        alias: 'verbose',
-        describe: 'be noisy - doesn not play well with -o',
-        type: 'boolean'
-    })
-    .help()
-    .alias('help', 'h')
-    .argv;
-
-var goggles = usb.findByIds("0x2ca3", "0x1f")
-
-if(!goggles) {
-    console.error("Goggles USB device not found. Please connect your goggles and restart the script.")
-    process.exit(1)
-}
-goggles.open()
-if(!goggles.interfaces) {
-    console.error("Couldn't open Goggles USB device")
+  .option("f", {
+    alias: "file",
+    describe: "output video feed to file",
+    type: "string",
+  })
+  .option("o", {
+    alias: "stdout",
+    describe:
+      "send video feed to stdout for playback. eg: node index.js -o | ffplay -",
+    type: "boolean",
+  })
+  .option("s", {
+    alias: "readsize",
+    describe: "size in bytes to queue for usb bulk interface reads",
+    default: 512,
+    type: "integer",
+  })
+  .option("q", {
+    alias: "queuesize",
+    describe: "number of polling usb bulk read requests to keep in flight",
+    default: 3,
+    type: "integer",
+  })
+  .option("v", {
+    alias: "verbose",
+    describe: "be noisy - doesn not play well with -o",
+    type: "boolean",
+  })
+  .help()
+  .alias("help", "h").argv
+
+// Main logic, needs async capability to use promises correctly
+const mainFunction = async () => {
+  let goggles = usb.findByIds("0x2ca3", "0x1f")
+
+  while (!goggles) {

```

Comment:
You can use `usb.on('attach', function(device) { ... });` to add an event listener for when a device is attached, save constantly polling and having a potential two second delay between attaching the googles and the script recognising them